File: 1.0/src/c/cz/mg/vulkan/PFN_vkDestroyInstance.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkDestroyInstance_call(JNIEnv* env, jclass jc, jlong address, jlong instance, jlong pAllocator){
    (void)env;
    (void)jc;
    PFN_vkDestroyInstance f = (PFN_vkDestroyInstance)jniLongToFunctionPointer(address);
    f(
        *((VkInstance*)jniLongToPointer(instance)),
        ((VkAllocationCallbacks*)jniLongToPointer(pAllocator))
    );
}

File: 1.0/src/cz/mg/vulkan/PFNvkDestroyInstance.java
package cz.mg.vulkan;

public class PFNvkDestroyInstance extends VkFunctionPointer {
    public PFNvkDestroyInstance() {
    }

    protected PFNvkDestroyInstance(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkDestroyInstance(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkDestroyInstance(long value) {
        setValue(value);
    }

    public PFNvkDestroyInstance(VkInstance instance) {
        super(instance, new VkString("vkDestroyInstance"));
    }

    public void call(VkInstance instance, VkAllocationCallbacks pAllocator){
        callNative(getValue(), instance != null ? instance.getVkAddress() : VkPointer.getNullAddressNative(), pAllocator != null ? pAllocator.getVkAddress() : VkPointer.NULL);
    }

    protected static native void callNative(long vkaddress, long instance, long pAllocator);

}
